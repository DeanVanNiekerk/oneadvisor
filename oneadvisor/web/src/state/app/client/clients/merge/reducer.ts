import { ClientEdit, Client } from "../types";
import { ClientMergeAction } from "./actions";

export type State = {
    readonly clients: Client[];
    readonly fetching: boolean;
    readonly currentStepIndex: number;
    readonly steps: string[];
    readonly insertedClient: ClientEdit | null;
};

export const defaultState: State = {
    clients: [],
    fetching: false,
    currentStepIndex: 0,
    steps: ["Clients", "Review", "Result"],
    insertedClient: null,
};

export const reducer = (state: State = defaultState, action: ClientMergeAction): State => {
    switch (action.type) {
        case "CLIENTS_MERGE_SOURCE_RECEIVE": {
            return {
                ...state,
                clients: action.payload.items,
                fetching: false,
            };
        }
        case "CLIENTS_MERGE_SOURCE_FETCHING": {
            return {
                ...state,
                fetching: true,
            };
        }
        case "CLIENTS_MERGE_SOURCE_FETCHING_ERROR": {
            return {
                ...state,
                clients: [],
                fetching: false,
            };
        }
        case "CLIENTS_MERGE_NEXT_STEP": {
            return {
                ...state,
                currentStepIndex: state.currentStepIndex + 1,
            };
        }
        case "CLIENTS_MERGE_PREVIOUS_STEP": {
            return {
                ...state,
                currentStepIndex: state.currentStepIndex - 1,
            };
        }
        case "CLIENTS_MERGE_RESET": {
            return {
                ...state,
                clients: [],
                currentStepIndex: 0,
                insertedClient: null,
            };
        }
        case "CLIENTS_MERGE_RECEIVE": {
            return {
                ...state,
                insertedClient: action.payload ? action.payload.tag : null,
                fetching: false,
            };
        }
        case "CLIENTS_MERGE_FETCHING": {
            return {
                ...state,
                fetching: true,
                insertedClient: null,
            };
        }
        case "CLIENTS_MERGE_FETCHING_ERROR": {
            return {
                ...state,
                fetching: false,
            };
        }
        case "CLIENTS_MERGE_VALIDATION_ERROR": {
            return {
                ...state,
                fetching: false,
            };
        }
        default:
            return state;
    }
};
