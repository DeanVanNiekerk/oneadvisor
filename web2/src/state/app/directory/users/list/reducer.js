// @flow

import * as types from './actions';
import type { User } from './types'
import type { Action } from './actions'

export type State = {
  +items: User[],
  +fetching: boolean,
  +error: boolean
}

export const defaultState: State = {
  items: [],
  fetching: false,
  error: false
}

export const reducer = (state: State = defaultState, action: Action) => {

  switch (action.type) {
    case "USERS_LIST_RECEIVE": {
      return {
        ...state,
        items: action.payload,
        fetching: false,
        error: false
      }
    }
    case "USERS_LIST_FETCHING": {
      return {
        ...state,
        fetching: true
      }
    }
    case "USERS_LIST_FETCHING_ERROR": {
      return {
        ...state,
        items: [],
        fetching: false,
        error: true
      }
    }
    default: return state;
  }

};
