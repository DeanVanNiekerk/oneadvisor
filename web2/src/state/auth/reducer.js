import * as types from './actions';

export const defaultState = {
  authenticated: false,
  userInfo: null,
  idToken: null,
  accessToken: null
}

export default (state = defaultState, action) => {

  switch (action.type) {
    case types.AUTH_RECIEVE_AUTHENTICATION: {
      return {
        ...state,
        authenticated: true,
        userInfo: action.payload.userInfo,
        idToken: action.payload.idToken,
        accessToken: action.payload.accessToken
      }
    }
    case types.AUTH_RECIEVE_AUTHENTICATION_CLEAR: {
      return {
        ...defaultState
      }
    }
    default: return state;
  }

};
