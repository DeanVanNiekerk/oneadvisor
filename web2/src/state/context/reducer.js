import { menus } from 'config/menu'
import { applications } from 'config/application'

export const defaultState = {
  applications: applications,
  menus: menus
}

export default (state = defaultState, action) => {

  switch (action.type) {
    default: return state;
  }

};