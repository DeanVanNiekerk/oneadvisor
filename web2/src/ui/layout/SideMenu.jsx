// @flow

import React, { Component } from "react";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import Icon from "@material-ui/core/Icon";
import DrawerMUI from "@material-ui/core/Drawer";
import Toolbar from "@material-ui/core/Toolbar";
import List from "@material-ui/core/List";
import Divider from "@material-ui/core/Divider";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import AppBar from "@material-ui/core/AppBar";
import ListSubheader from "@material-ui/core/ListSubheader";

import type { State as RootState } from "state/rootReducer";
import type { Menu, Application } from "state/context/types";
import {
  currentMenuSelector,
  currentApplicationSelector
} from "state/context/selectors";

// const Wrapper = styled.div`
//     order: 0;
//     border-right: 3px solid #d4d4d4;
//     background-color: #f0f0f0;
//     position: sticky !important;
//     top: 3.5rem;
//     z-index: 1000;
//     height: calc(100vh - 3.7rem);
//     overflow-y: auto;
//     padding-top: 1rem;

//     font-size: 0.9rem;

//     -webkit-box-shadow: inset -5px 0px 5px -4px rgba(212,212,212,1);
//     -moz-box-shadow: inset -5px 0px 5px -4px rgba(212,212,212,1);
//     box-shadow: inset -5px 0px 5px -4px rgba(212,212,212,1);
// `;

// const NavItem = styled(NavItemBs)`
//     ${props => props.link.isCurrent && css`
//         background-color: #d4d4d4;
//         border-left: 6px solid ${props => props.application.color};
//     `}
//     ${props => !props.link.isCurrent && css`
//         border-left: 6px solid #f0f0f0;
//     `}
// `;

const drawerWidth = 240;

const Drawer = styled(DrawerMUI)`
  width: ${drawerWidth}px !important;
  flex-shrink: 0 !important;
`;
const styles = () => ({
  drawerPaper: {
    width: drawerWidth
  }
});

type Props = {
  className: string,
  menu: Menu,
  application: Application,
  classes: Object
};

class SideMenu extends Component<Props> {
  render() {
    return (
      <Drawer
        variant="permanent"
        classes={{
          paper: this.props.classes.drawerPaper
        }}
      >
        <AppBar position="static">
          <Toolbar />
        </AppBar>

        {this.props.menu.groups.map(group => {
          return (
            <List
              key={group.name}
              subheader={<ListSubheader>{group.name}</ListSubheader>}
            >
              {group.links.map(link => {
                return (
                  <ListItem
                    key={link.relativePath}
                    application={this.props.application}
                    link={link}
                  >
                    <ListItemIcon>
                      <Icon>{link.icon}</Icon>
                    </ListItemIcon>
                    <ListItemText
                      to={`${this.props.menu.relativePath}${link.relativePath}`}
                    >
                      {link.name}
                    </ListItemText>
                  </ListItem>
                );
              })}
            </List>
          );
        })}
      </Drawer>
      // <Wrapper>

      //     {this.props.menu.groups.map(group => {

      //         return (<div key={group.name}>
      //             <div className="pl-2 pb-1 text-uppercase">{group.name}</div>
      //             <Nav className="flex-column">

      //                 {group.links.map(link => {
      //                     return (
      //                         <NavItem key={link.relativePath} className="pl-3" application={this.props.application} link={link}>
      //                             <Icon className="float-left mt-2 text-dark">{link.icon}</Icon>
      //                             <Link className="float-left nav-link text-dark pl-2" to={`${this.props.menu.relativePath}${link.relativePath}`}>
      //                                 {link.name}
      //                             </Link>
      //                         </NavItem>
      //                     )})
      //                 }

      //             </Nav>
      //         </div>
      //         )
      //     })}

      // </Wrapper>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  menu: currentMenuSelector(state),
  application: currentApplicationSelector(state)
});

export default connect(mapStateToProps)(withStyles(styles)(SideMenu));
