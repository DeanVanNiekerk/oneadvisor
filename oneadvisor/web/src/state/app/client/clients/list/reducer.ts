import { Filters, PageOptions, SortOptions } from "@/app/table";
import { defaultPageOptions, defaultSortOptions } from "@/app/table/defaults";

import { Client } from "../types";
import { ClientListAction } from "./actions";

export type State = {
    readonly items: Client[];
    readonly totalItems: number;
    readonly fetching: boolean;
    readonly pageOptions: PageOptions;
    readonly sortOptions: SortOptions;
    readonly filters: Filters | null;
    readonly selectedClientIds: string[];
};

export const defaultState: State = {
    items: [],
    totalItems: 0,
    fetching: false,
    pageOptions: defaultPageOptions(),
    sortOptions: defaultSortOptions("", "desc"),
    filters: null,
    selectedClientIds: [],
};

export const reducer = (state: State = defaultState, action: ClientListAction): State => {
    switch (action.type) {
        case "CLIENTS_LIST_RECEIVE": {
            return {
                ...state,
                totalItems: action.payload.totalItems,
                items: action.payload.items,
                fetching: false,
            };
        }
        case "CLIENTS_LIST_FETCHING": {
            return {
                ...state,
                fetching: true,
            };
        }
        case "CLIENTS_LIST_FETCHING_ERROR": {
            return {
                ...state,
                items: [],
                fetching: false,
            };
        }
        case "CLIENTS_LIST_PAGE_OPTIONS_RECEIVE": {
            return {
                ...state,
                pageOptions: {
                    ...action.payload,
                },
            };
        }
        case "CLIENTS_LIST_SORT_OPTIONS_RECEIVE": {
            return {
                ...state,
                sortOptions: {
                    ...action.payload,
                },
            };
        }
        case "CLIENTS_LIST_FILTERS_RECEIVE": {
            return {
                ...state,
                filters: {
                    ...action.payload,
                },
            };
        }
        case "CLIENTS_LIST_SELECTED_RECEIVE": {
            return {
                ...state,
                selectedClientIds: action.payload,
            };
        }
        default:
            return state;
    }
};
