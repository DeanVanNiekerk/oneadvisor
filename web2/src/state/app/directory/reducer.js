import { combineReducers } from 'redux';

import { reducer as users } from './users/reducer';

export const reducer = combineReducers({
  users,
});