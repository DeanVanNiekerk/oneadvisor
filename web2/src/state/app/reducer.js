import { combineReducers } from 'redux';
import { reducer as directory } from './directory/reducer';

export default combineReducers({
  directory,
});
