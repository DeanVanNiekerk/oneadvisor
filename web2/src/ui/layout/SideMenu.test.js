// @flow

import React from 'react';
import SideMenu from './SideMenu';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom' 
import configureMockStore from 'redux-mock-store'
import { defaultState as defaultConextState} from 'state/context/reducer'

const mockStore = configureMockStore()

describe('SideMenu', () => {

    it('renders correctly', () => {

        const store = mockStore({
            context: {
                ...defaultConextState
            },
            router: {
                location: {
                    pathname: ""
                }
            }
        })

        const tree = renderer
            .create(<Router><SideMenu store={store} /></Router>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

});