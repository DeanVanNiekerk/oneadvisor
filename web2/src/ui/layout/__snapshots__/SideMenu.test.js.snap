// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu renders correctly 1`] = `
<div
  className="sc-bdVaJa eqEQhl"
>
  <div>
    <div
      className="pl-2 pb-1 text-uppercase"
    >
      Management
    </div>
    <ul
      className="flex-column nav"
    >
      <li
        application={
          Object {
            "color": "#3949ab",
            "icon": "security",
            "id": "DIRECTORY",
            "isCurrent": true,
            "name": "Directory",
            "relativePath": "/directory",
          }
        }
        className="pl-3 sc-bwzfXH kvaeoU nav-item"
        link={
          Object {
            "icon": "people",
            "isCurrent": true,
            "isDefault": true,
            "name": "Users",
            "relativePath": "/users",
          }
        }
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-1 float-left mt-2 text-dark"
        >
          people
        </span>
        <a
          className="float-left nav-link text-dark pl-2"
          href="/directory/users"
          onClick={[Function]}
        >
          Users
        </a>
      </li>
      <li
        application={
          Object {
            "color": "#3949ab",
            "icon": "security",
            "id": "DIRECTORY",
            "isCurrent": true,
            "name": "Directory",
            "relativePath": "/directory",
          }
        }
        className="pl-3 sc-bwzfXH hpXLxf nav-item"
        link={
          Object {
            "icon": "business",
            "isCurrent": false,
            "isDefault": false,
            "name": "Organisations",
            "relativePath": "/organisations",
          }
        }
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-1 float-left mt-2 text-dark"
        >
          business
        </span>
        <a
          className="float-left nav-link text-dark pl-2"
          href="/directory/organisations"
          onClick={[Function]}
        >
          Organisations
        </a>
      </li>
    </ul>
  </div>
</div>
`;
